### PR TITLE
Fix branch commit message too long problem

### PR DESCRIPTION
**Pull request by lunny**
When branch's commit CommitMessage is too long, the column maybe too short.(TEXT 16K for mysql).
This PR will fix it to only store the summary because these message will only show on branch list or possible future search?

### CODE REVIEW

**Comment by wolfogre** [comment]
I am sure it works, but it's unclear when to use message summary.
image

My suggestion:
https://github.com/go-gitea/gitea/compare/main...wolfogre:gitea:bugfix/fix_branch_commit_message_len

Now it's clear:

image

**Review by wolfogre — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
@lunny backport?

File: modules/repository/branch.go [hunk 1]
```diff
@@ -77,19 +77,19 @@ func SyncRepoBranchesWithRepo(ctx context.Context, repo *repo_model.Repository,
 				RepoID:        repo.ID,
 				Name:          branch,
 				CommitID:      commit.ID.String(),
-				CommitMessage: commit.CommitMessage,
```
Comment (wxiaoguang) [inline modules/repository/branch.go:80]: The meaning of "CommitMessage" in the branch struct has changed now.

If it's in main dev branch so there is still a chance to rename the field.

At least, there should be comment to say that it only contains the message summary but not the full message and why.

**Comment by wxiaoguang** [comment]
And, I am curious what why it's merged too fast. AFAIK,  silverwind also complained that some PRs got merged too fast and suggested to wait for enough time after lgtm/done

**Comment by lunny** [comment]
> @lunny backport?

It cannot be backport because branch is 1.21 only feature

**Pull request merged**